alacritty_terminal/src/config/bindings.rs

```diff
@@ -71,113 +71,158 @@ pub fn default_key_bindings() -> Vec<KeyBinding> {
+            Action::Esc(format!("\x1b[1;{}D", modifiers_code));
+            Key::F1,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[1;{}P", modifiers_code));
+            Key::F2,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[1;{}Q", modifiers_code));
+            Key::F3,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[1;{}R", modifiers_code));
+            Key::F4,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[1;{}S", modifiers_code));
+            Key::F5,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[15;{}~", modifiers_code));
+            Key::F6,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[17;{}~", modifiers_code));
+            Key::F7,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[18;{}~", modifiers_code));
+            Key::F8,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[19;{}~", modifiers_code));
+            Key::F9,    [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[20;{}~", modifiers_code));
+            Key::F10,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[21;{}~", modifiers_code));
+            Key::F11,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[23;{}~", modifiers_code));
+            Key::F12,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[24;{}~", modifiers_code));
+            Key::F13,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[25;{}~", modifiers_code));
+            Key::F14,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[26;{}~", modifiers_code));
+            Key::F15,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[28;{}~", modifiers_code));
+            Key::F16,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[29;{}~", modifiers_code));
+            Key::F17,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[31;{}~", modifiers_code));
+            Key::F18,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[32;{}~", modifiers_code));
+            Key::F19,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[33;{}~", modifiers_code));
+            Key::F20,   [shift: mods.shift, alt: mods.alt, ctrl: mods.ctrl];
+            Action::Esc(format!("\x1b[34;{}~", modifiers_code));
+        ));
+
+        if modifiers_code != 2 {
```

Comment:
I think a comment explaining why we do things differently here would go a long way. Otherwise it would probably strike people as odd.

Reply:
Could we give names to each of these magic numbers from the table with a bitflags struct? I'd much prefer to be comparing like `if modifiers_code != ModifierCode::Shift`. It might clean up the iteration too a little bit, IDK.

Reply:
It would be nice, if `2 | 3 == 4` and other bitwise math would map on the codes, but that's not the case, so the array + table in a comment should do the job right. I also think that's too much hassle for the code that will likely remain untouched anymore.

Reply:
I've added a comment on why we're skipping things here btw

Reply:
OK, yea I tried for a moment to make something nicer and it's indeed probably not worth it. However I think the comment should address the main difference I see, which is that in the bindings defined above, there's explicit actions, and conditions for the `mode`. I think ideally we'd just overwrite them, but since I don't think that's easy to do ATM, so this is fine.